platform/security/idp/security-idp-client/src/main/java/org/codice/ddf/security/idp/client/LogoutRequestService.java

```diff
@@ -375,28 +389,37 @@ private String getEntityId() {
     return String.format("https://%s:%s%s/saml", hostname, port, rootContext);
   }
 
-  private void logout() {
-    HttpSession session = sessionFactory.getOrCreateSession(request);
+  private SecurityAssertion getIdpSecurityAssertion() {
 
-    SecurityTokenHolder tokenHolder =
-        ((SecurityTokenHolder) session.getAttribute(SecurityConstants.SAML_ASSERTION));
+    return new SecurityAssertionImpl(getTokenHolder().getSecurityToken(IDP_REALM_NAME));
+  }
 
-    SecurityAssertion securityAssertion =
-        new SecurityAssertionImpl(tokenHolder.getSecurityToken("idp"));
+  private HttpSession getOrCreateSession() {
+    return sessionFactory.getOrCreateSession(request);
+  }
```

Comment:
New helper method `getOrCreateSession`. Take advantage of fact that `request` is an object attribute.